app/services/doc_auth/lexis_nexis/request.rb

```diff
@@ -146,13 +147,14 @@ def hmac_authorization
       end
 
       def settings
+        @reference = uuid
```

Comment:
can we do `@referece = uuid unless defined?(@reference)`, so that we will only generate uuid once for this object, otherwise this function is not idempotent and may have unwanted consequences.

Reply:
> can we do `@referece = uuid unless defined?(@reference)`, so that we will only generate uuid once for this object, otherwise this function is not idempotent and may have unwanted consequences when called multiple times.

as per our chat ... this suggestion would change the current functionality